src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -752,6 +791,96 @@ protected function compileEndpush($expression)
     }
 
     /**
+     * Compile the inject annotations into valid PHP.
+     *
+     * @param  string[] $segments
+     *
+     * @return string
+     */
+    protected function compileInjectAnnotation(array $segments)
+    {
+        list($injectClass, $variableName) = $segments;
+
+        return '<?php '.$variableName.' = app('.$injectClass.'::class); ?>';
```

Comment:
Since the syntax has the type as an identifier like `$var instanceof SomeIdentifier` that would be expected behaviour no? The `@param` annotation behaves the same with class names.